src/PelicanFile.cc

```diff
@@ -44,23 +100,32 @@ File::Open(const std::string      &url,
         return XrdCl::XRootDStatus(XrdCl::stError, XrdCl::errNotSupported);
     }
 
+    m_header_timeout.tv_nsec = m_default_header_timeout.tv_sec;
+    m_header_timeout.tv_sec = m_default_header_timeout.tv_nsec;
+    auto pelican_url = XrdCl::URL();
+    pelican_url.SetPort(0);
+    if (!pelican_url.FromString(url)) {
+        m_logger->Error(kLogXrdClPelican, "Failed to parse pelican:// URL as a valid URL");
+        return XrdCl::XRootDStatus(XrdCl::stError, XrdCl::errInvalidArgs);
+    }
+    auto pm = pelican_url.GetParams();
+    auto iter = pm.find("pelican.timeout");
+    std::string header_value = (iter == pm.end()) ? "" : iter->second;
+    m_header_timeout = GetTimeoutFromHeader(header_value, m_logger);
```

Comment:
I'm a bit confused about this section. The function "GetTimeoutFromHeader" makes it seem like you want to construct and return a timespec using some information from a header, but this only looks at a value provided via URL parameters as far as I can tell. I thought maybe `XrdCl::URL::GetParams()` did something fancy to check headers as well, but poking around the source for that makes it look like it only works against URL params.

Does this not implement a way to grab a timeouts headers? The PR description mentions `X-Pelican-Timeout`, but I don't see that defined anywhere in the code, and I don't see where we'd be checking for it.

Reply:
You're right, it is confusing.

XRootD takes the headers we want to capture and adds it to the URL's query string.  So, even though I think of these as headers (and that's how they originate), in the context in the code, they're query string parameters.

I'll refactor a bit to clarify.